packages/block-library/src/search/index.php

```diff
@@ -29,11 +29,16 @@ function render_block_core_search( $attributes ) {
 		)
 	);
 
-	$input_id            = wp_unique_id( 'wp-block-search__input-' );
-	$classnames          = classnames_for_block_core_search( $attributes );
-	$show_label          = ! empty( $attributes['showLabel'] );
-	$use_icon_button     = ! empty( $attributes['buttonUseIcon'] );
-	$show_button         = ( ! empty( $attributes['buttonPosition'] ) && 'no-button' === $attributes['buttonPosition'] ) ? false : true;
+	$input_id        = wp_unique_id( 'wp-block-search__input-' );
+	$classnames      = classnames_for_block_core_search( $attributes );
+	$show_label      = ( ! empty( $attributes['showLabel'] ) ) ? true : false;
+	$use_icon_button = ( ! empty( $attributes['buttonUseIcon'] ) ) ? true : false;
+	$show_button     = true;
+	if ( isset( $block->context['enhancedPagination'] ) && $block->context['enhancedPagination'] ) {
```

Comment:
Should this check be limited to the situation where the Gutenberg experiment is enabled?

Reply:
ok, done in 97b3c845e11c0d1394949e55488a38a2d5eb1353